### ISSUE TITLE
Ripped MPEG2 DVD VIDEO_TS folder with .IFO and .VOB files playback freeezes and skips on Raspberry PI 2

### ISSUE THREAD

**Issue by awtoner** [xbmc/xbmc#15966]
### Describe the bug
Here is a clear and concise description of what the problem is:

A long time ago I ripped DVD's to my NAS using AnyDVD. Problems like this have been going on for years with XBMC/Kodi, getting better/worse as releases come and go. With 18.2-RC1, I'm back to not enjoying the Simpsons due to playback issues. After playing for a few minutes, repeatably around the same time in the video, playback will either freeze for a few minutes and/or start skipping ahead and lose audio sync with the video.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Wife and kids should be able to play a DVD, from a folder and with menus, with hardware acceleration and de-interlacing, without any playback problems.

## Actual Behavior

I can navigate the DVD menu and start playback just fine. Several minutes into the episode, the video will either freezer or it will start skipping ahead and the audio will lose sync with the video. After a few minutes it'll come back, perhaps with audio sync problems continuing if the video froze. Rewinding back to the time of the problem is usually impossible. I can reproduce the problem at about the same time, e.g. around 8 minutes into season 7, episode 8.

The last time I investigates this a year or two ago, Kodi/XMBC on Windows would play the DVD folders just fine but on my Raspberry Pi 2 devices it would skip. I haven't tried this with Kodi 18.2.

## Possible Fix

Is there a set-top box other than Kodi, OSMC, Raspberry PI 2, etc. that I should use to improve stability for DVD folder playback?
I want to play the full DVD with menus to retain special features and to avoid having to further process every episode into a standalone video.

### To Reproduce

1.  DVD's ripped as folders
2. TV show source already setup in Kodi
3. Play an episode, the DVD menu comes up, select the episode to play
4. Wait until the problem occurs or skip / fast forward to get closer
5. Video will freeze or skip

### Debuglog

The debuglog can be found here:
https://paste.ubuntu.com/p/npH4wnr3Sb/

Search the log for "simpsons" to see when it starts to play. I think it was around 23:45 to 23:52 when the problem happened. This time it was just a freeze for a few minutes.

BTW - It is really a pain in the to post a log. After jumping through the hoops to enable logging and upload the log file, I got an "Error posting the kodi.log file. The file is too large. Max size is 1 MB." I then had to use the file manager.

### Screenshots 
Here are some links or screenshots to help explain the problem:

None

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

CPU usage is around 25% per core or just less than 100% total while DVD is playing (hardware acceleration off)
Memory usage is around 19%.

I've tried changing several settings:
De-interlacing to none
Disabling both MAML and OMX hardware acceleration, enabling one or the other, enabling both
Sync playback to display, usually off, tried on
Other settings over the years including NFS and SMB, skip DVD intro, adjust display refresh rate, etc.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [X] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: OSMC 2019.04-1
 - Kodi version: 18.2-RC1

I tried creating a fresh profile but got the same problem.

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
As you are using OSMC and are asking for: 

>Is there a set-top box other than Kodi, OSMC, Raspberry PI 2, etc.

it would be nice if you could give LibreELEC on your RPi2 a shot. 

https://libreelec.tv/downloads_new/raspberry-pi-2/

and if you are still facing stuttering issues it would also be nice if you could share a log then. 

I'm not saying that it will solve your problem, but OSMC is patching the code a little bit as well as LibreELEC does. But LibreELEC does it in a different way, AFAIK. So we would appreciate if you could test LibreELEC and let us know if that improves something, 

Thanks in advance.

**Comment by popcornmix** [comment]
Testing on LE could be interesting.
Can you play the video locally (copy it to sdcard or USB stick)? That would rule out network issues.
Can you try ripping the file to mkv (without reencoding)? You would lose the menus, but would be interesting if that format plays correctly.

**Comment by samnazarko** [comment]
@DaVukovic: both osmc and LE use @popcornmix’s RBP backports branch, so the results should be identical. It’s worth testing though.

**Comment by samnazarko** [comment]
You can upload debug logs via My OSMC (max size is 10M)

**Comment by lharms** [comment]
I have seen similar behavior on other seasons of this show but with ISO and windows.  I suspect there is something in the structure of these DVDs that causes it as none of my other ISOs seem to do this.  I have unfortunately not been able to reliably to get it to happen and it happens randomly.  Is it just the simpsons or do other shows/movies do it too?  I tried at about the same spot as you on my copy ~9mins 6 seconds in and it took a couple of tries.  But I got it to freeze up.

In that log you can see the dysnc and the player is trying to correct it.  Basically getting a sync error over and over.  Then as the sync time gets lower and lower it eventually catches up and then corrects.

I suspect this is core and not linux/pi/network specific (as I am on windows).

Not totally sure what the code in xbmc\cores\AudioEngine\Engines\ActiveAE\ActiveAE.cpp around line 1931 is doing.  It looks like it is trying to put it back in sync but picking the wrong number to do so.  In your log you can see it is also going thru CVideoPlayer::CheckContinuity and trying to also correct it by going thru case 2 on line 2312.  This is eerily the same sort of thing I had on a bluray a while back. It is like it suddenly thinks the current spot is the beginning of the vob (at least on my run on my computer).  Then tries to bring it back into sync which causes all sorts of skipping.

I turned up the logging on mine
2019-04-23 21:25:20.637 T:3132   DEBUG: CVideoPlayer::CheckContinuity - resync backward :2, prev:547077344.444444, curr:229913000.000000, diff:-317164344.444444
2019-04-23 21:25:20.637 T:3132   DEBUG: CVideoPlayer::ProcessSubData: Got complete SPU packet
2019-04-23 21:25:20.637 T:3132   DEBUG: CVideoPlayer::CheckContinuity - resync backward :1, prev:546853677.777778, curr:229687966.666667, diff:-317165711.111111
2019-04-23 21:25:20.637 T:3132   DEBUG: CVideoPlayer::CheckContinuity - update correction: -317197711.111111
2019-04-23 21:25:20.651 T:7388 WARNING: ActiveAE - large audio sync error: 308984.009200
2019-04-23 21:25:20.655 T:9724   DEBUG: CRenderManager::PrepareNextRender: frameOnScreen: 229931320.400000 renderPts: 229997987.071753 nextFramePts: 538969244.000000 -> diff: -308971256.928247  render: 0 forceNext: 0

You can see the clock jumps way backwards.  There is that DVDNAV_CELL_CHANGE wonder if it is doing something cute.

**Comment by awtoner** [comment]
Thanks for the comments and suggestions. Given the banter in https://github.com/xbmc/xbmc/issues/15405 I was worried I wasn't going to get a warm reception.  It is a relief that lharms also has this problem with the Simpsons - this suggests it isn't something unique to my settings or rip. The family hasn't complained about DVD movies that I've ripped for them, so maybe it is just a Simpsons problem.

Here are my results:

Reproduced problem again this morning - freezes around 8:30 in s07e08. I skip to the 3rd chapter (5:43) then fast forward to around 8:30, or else just fast forward. When it freezes, the Kodi system may become unresponsive for 10's of seconds. Hitting stop, menu, and exit may have no effect, but I think this is inconsistent.

VLC on Windows 10 - no problems playing from the DVD folder on my NAS

Kodi on Windows 10 - This worked fine a year or two ago - I remember Windows not having a problem with the Simpsons when RPi did, but today I was having a problem getting the "disc" to play at all. I navigated to the VIDEO_TS.IFO, played it to get to the DVD menu, then played the episode. It froze up around the same time.

MKV - I used MakeMKV to create an MKV and stored it on the NAS. The problem is different on the RPi2 with the MKV than the DVD folder. It doesn't freeze or skip ahead (at least playing about 3 minutes around the time at which the DVD folder has a problem) but it did freeze when fast forwarding at 16x and 32x. At 8x I was able to get near the usual problem time and play through it without a hitch.

Local USB drive - I copied the DVD folder to a Windows-formatted USB stick, popped it in the RPi2, and played from the file manager. It froze at the usual time.

LibreELEC - I'm saving this for last since I'll need to get my IR remote setup again. I'm using a GPIO receiver and did some customization with my Harmony remote, and I remember remote setting being a pain every time I've switched distributions. I read a bit about IR remotes on LibreELEC, and it looks like I could run into some of this again. Maybe I can just use the Kodi remote app on my smartphone.

**Comment by awtoner** [comment]
Same problem on a fresh installation of LibreELEC-RPi2.arm-9.0.1. The Simpsons s07e08 froze around 8:30 again with the same procedure to skip to chapter 3 then fast forward as described above. The video stayed frozen for at least three minutes, at which point I gave up, hit stop, and went back to get the log file.
https://paste.ubuntu.com/p/YnG8bMSJzF/

**Comment by popcornmix** [comment]
If you try playing the VOB files inside VIDEO_TS do you get the same hang?
Are you able to produce a cut down sample file that exhibits the problem?

As the problem occurs on Windows (and presumably all platforms) then getting a sample file
so devs can reproduce the problem is probably the best way of getting a fix.

**Comment by awtoner** [comment]
I'm attempting to make a cut-down video file that exhibits the problem, but am running into a few problems:
- I made a cut-down file with Avidemux, trying to preserve/copy as much as possible instead of transcoding. I replaced the VTS_02_2.VOB file on my NAS with the cut-down one, but when I try to play the episode from the DVD menu on my RPi it exits back to the Kodi main menu. I think I got this file to play on Kodi on Windows 10 last week, but it didn't reproduce the problem.
- Any better suggestions than Avidemux?
- Kodi 18.2 on Windows 10 won't play the Simpsons TV shows from the TV shows menu. I can navigate to the season and episode, but when I try to play the video there's just a brief flash and no playing. I'm instead going through the File manager. Maybe there's already a bug report on this.
- Kodi on Window also looses fast forward and scrolling the progress bar for some video files - again maybe there's already a bug on that one. This makes it hard to get 8 minutes into the video.

Thanks for the ongoing help.

Adam
